src/dotnet-scaffolding/dotnet-scaffold-aspnet/ScaffolderBuilderExtensions.cs

```diff
@@ -0,0 +1,99 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System;
+using System.IO;
+using System.Linq;
+using Microsoft.DotNet.Scaffolding.Helpers.Services;
+using Microsoft.DotNet.Scaffolding.TextTemplating;
+using Microsoft.DotNet.Tools.Scaffold.AspNet.Commands.Blazor.BlazorCrud;
+using Microsoft.DotNet.Tools.Scaffold.AspNet.Commands.MinimalApi;
+using Microsoft.DotNet.Tools.Scaffold.AspNet.Helpers;
+
+namespace Microsoft.DotNet.Scaffolding.Core.Builder;
+
+internal static class ScaffolderBuilderExtensions
+{
+    public static IScaffoldBuilder WithMinimalApiTextTemplatingStep(this IScaffoldBuilder builder)
+    {
+        var allT4Templates = new TemplateFoldersUtilities().GetAllT4Templates(["MinimalApi"]);
+        string? t4TemplatePath = null;
+        return builder.WithStep<TextTemplatingStep>(config =>
+        {
+            var step = config.Step;
+            var context = config.Context;
+            MinimalApiModel? minimalApiModel = null;
+            if (context.Properties.TryGetValue(nameof(MinimalApiModel), out var minimalApiModelObj) &&
+                minimalApiModelObj is MinimalApiModel)
+            {
+                minimalApiModel = minimalApiModelObj as MinimalApiModel;
+            }
+
+            ArgumentNullException.ThrowIfNull(minimalApiModel);
+            if (minimalApiModel.DbContextInfo.EfScenario)
+            {
+                t4TemplatePath = allT4Templates.FirstOrDefault(x => x.EndsWith("MinimalApiEf.tt", StringComparison.OrdinalIgnoreCase));
+            }
+            else
+            {
+                t4TemplatePath = allT4Templates.FirstOrDefault(x => x.EndsWith("MinimalApi.tt", StringComparison.OrdinalIgnoreCase));
+            }
+
+            var templateType = MinimalApiHelper.GetMinimalApiTemplateType(t4TemplatePath);
+
+            if (string.IsNullOrEmpty(t4TemplatePath) ||
+                string.IsNullOrEmpty(minimalApiModel.EndpointsPath) ||
+                templateType is null)
+            {
+                throw new InvalidOperationException("could not get minimal api template");
+            }
+
+            step.TemplatePath = t4TemplatePath;
+            step.TemplateType = templateType;
+            step.TemplateModel = minimalApiModel;
+            step.TemplateModelName = "Model";
+            step.OutputPath = minimalApiModel.EndpointsPath;
+        });
+        
+    }
+
+    public static IScaffoldBuilder WithBlazorCrudTextTemplatingStep(
+        this IScaffoldBuilder builder)
+    {
+        var allT4TemplatePaths = new TemplateFoldersUtilities().GetAllT4Templates(["BlazorCrud"]);
```

Comment:
Same question as above

Reply:
easier in the `WithMinimalApiTextTemplatingStep` but harder here since we are returning builder with multiple steps. Unless I can add steps to the builder from within an WithStep pre execute block. Any thoughts?

Reply:
Ah yes, you're right, my bad. We may want to consider looking at this again later (perhaps some way to have the Step itself expand this and iterate within itself?) to avoid the perf hit on get-command, --version, etc calls. But you're right I misread it as being similar to the last one.

Reply:
yes I will definitely want to rework this but couldn't come up with anything better rn. I'll sit with and a TODO to make this better.